### PR TITLE
CVE-2005-3357

### PR DESCRIPTION
**Pull request by Dmj1097**
This is the updated document with information relevant to all aspects of the NPE through sending a non-SSL request to a SSL port, allowing for DOS attack, vulnerability.

### CODE REVIEW

File: cves/CVE-2005-3357.yml [hunk 1]
```diff
@@ -31,7 +31,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  There was a component of httpd where, if the user had a specific type of SSL
```
Comment (Ksxerity) [inline cves/CVE-2005-3357.yml]: Viewers may not know what SSL is or stands for.

File: cves/CVE-2005-3357.yml [hunk 2]
```diff
@@ -40,8 +45,8 @@ bounty:
   amt:
   announced:
   url:
-reviews: []
-bugs: []
+reviews: [354394]
+bugs: [37791]
 repo:
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
```
Comment (Ksxerity) [inline cves/CVE-2005-3357.yml:53]: Even if the commits were provided for you, you should still look at the commits and make some kind of a note.

File: cves/CVE-2005-3357.yml [hunk 3]
```diff
@@ -94,11 +102,14 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer:
-  date:
-  automated:
-  google:
-  contest:
+  answer: |
+    This answer appears to have been found by a random user who discovered the
```
Comment (Ksxerity) [inline cves/CVE-2005-3357.yml]: I think you mean "This vulnerability"

File: cves/CVE-2005-3357.yml [hunk 4]
```diff
@@ -166,17 +189,26 @@ lessons:
...
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+      The NPE comes from a non-ssl request being sent through an ssl port. The
+      vulnerability would have been preventable if additional checks were in
+      place to make sure no non-ssl request could be sent to an ssl port, or if
+      verification would happen before the local variable "ssl" was reffered to
+      when it was still nul
```
Comment (Ksxerity) [inline cves/CVE-2005-3357.yml]: "nul" should be "null."

File: cves/CVE-2005-3357.yml [hunk 5]
```diff
@@ -166,17 +189,26 @@ lessons:
...
     note:
   environment_variables:
-    applies:
-    note:
+    applies: true
+    note: |
+      This vulnerability is based entirely on the fact that the interaction
+      between when a request was sent in and when the handler attempts to access
+      the ssl hook, is done with the same variable and there is no handling for
+      when the current ssl is nul.
```
Comment (itskyleki) [inline cves/CVE-2005-3357.yml]: "nul" should be "null" again

File: cves/CVE-2005-3357.yml [hunk 6]
```diff
@@ -199,4 +231,15 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    The main mistake with this vulnerability was the lack of handling when ssl
+    was nul. The team never accounted for a non-ssl request to be made on a ssl
+    port when if ssl is not currently in use. They seem to have had it written
```
Comment (Ksxerity) [inline cves/CVE-2005-3357.yml:240]: Decide between "when" or "if".

**Comment by Ksxerity** [comment]
Interesting to read. +One up vote.

**Comment by kleenip** [comment]
1 upvote. Seems to be mostly the team not accounting for potential errors.

**Pull request merged**